File: go/internal/handlers/roles.go
package handlers

import (
	"net/http"
	"strings"

	"github.com/go-chi/chi/v5"
)

var roleChoices = []struct {
	Code string `json:"code"`
	Name string `json:"name"`
}{
	{"ADMIN", "Administrator"},
	{"SHOPKEEPER", "Shopkeeper"},
	{"CUSTOMER", "Customer"},
}

// ListRoles returns all roles (Bolt-compatible).
func ListRoles(w http.ResponseWriter, r *http.Request) {
	writeJSON(w, http.StatusOK, roleChoices)
}

// GetRoleByCode returns role by code.
func GetRoleByCode(w http.ResponseWriter, req *http.Request) {
	code := strings.TrimSpace(strings.ToUpper(chi.URLParam(req, "code")))
	for _, role := range roleChoices {
		if role.Code == code {
			writeJSON(w, http.StatusOK, role)
			return
		}
	}
	writeJSON(w, http.StatusNotFound, map[string]string{"detail": "Role not found"})
}

File: express/src/routes/roles.js
/**
 * Role routes: /roles, /roles/code/:code (Bolt-compatible).
 */

import { Router } from "express";
import { ROLE_CHOICES } from "../config.js";

const router = Router();

router.get("/roles", (req, res) => {
  res.json(ROLE_CHOICES.map((r) => ({ code: r.code, name: r.name })));
});

router.get("/roles/code/:code", (req, res) => {
  const code = (req.params.code || "").trim().toUpperCase();
  const role = ROLE_CHOICES.find((r) => r.code === code);
  if (!role) {
    return res.status(404).json({ detail: "Role not found" });
  }
  res.json({ code: role.code, name: role.name });
});

export default router;
